transport/http/outbound.go

```diff
@@ -111,30 +152,35 @@ func (o outbound) Call(ctx context.Context, req *transport.Request) (*transport.
 	deadline, _ := ctx.Deadline()
 	ttl := deadline.Sub(start)
 
-	request, err := http.NewRequest("POST", o.URL, req.Body)
+	req, err := http.NewRequest("POST", o.URL, treq.Body)
 	if err != nil {
 		return nil, err
 	}
 
-	request.Header = applicationHeaders.ToHTTPHeaders(req.Headers, nil)
-	if hs := baggage.FromContext(ctx); hs.Len() > 0 {
-		request.Header = baggageHeaders.ToHTTPHeaders(hs, request.Header)
-	}
+	ctx, span := o.createSpan(ctx, req, treq, start)
+	defer span.Finish()
 
-	request.Header.Set(CallerHeader, req.Caller)
-	request.Header.Set(ServiceHeader, req.Service)
-	request.Header.Set(ProcedureHeader, req.Procedure)
-	request.Header.Set(TTLMSHeader, fmt.Sprintf("%d", ttl/time.Millisecond))
+	req.Header.Set(CallerHeader, treq.Caller)
+	req.Header.Set(ServiceHeader, treq.Service)
+	req.Header.Set(ProcedureHeader, treq.Procedure)
+	req.Header.Set(TTLMSHeader, fmt.Sprintf("%d", ttl/time.Millisecond))
 
-	encoding := string(req.Encoding)
+	encoding := string(treq.Encoding)
 	if encoding != "" {
-		request.Header.Set(EncodingHeader, encoding)
+		req.Header.Set(EncodingHeader, encoding)
```

Comment:
This is not checked elsewhere. This code should only be called by an encoding layer and it is the encoding’s responsibility to set this property, but this is a public interface. Encodings can also be implemented outside the purview of YARPC.